internal/cli/cmd/login/login.go

```diff
@@ -177,14 +177,13 @@ func (o *loginOptions) run(ctx context.Context) error {
 		refreshToken = ""
 	}
 
-	if err = libConfig.SaveCLIConfig(
-		libConfig.CLIConfig{
-			APIAddress:            o.ServerAddress,
-			BearerToken:           bearerToken,
-			RefreshToken:          refreshToken,
-			InsecureSkipTLSVerify: o.InsecureTLS,
-		},
-	); err != nil {
+	// Merge new settings with the existing configuration
+	o.Config.APIAddress = o.ServerAddress
+	o.Config.BearerToken = bearerToken
+	o.Config.RefreshToken = refreshToken
+	o.Config.InsecureSkipTLSVerify = o.InsecureTLS
```

Comment:
This looks very close, but I think that if the server we're logging into isn't the same as the one we were previously logged into, then we _don't_ want to merge because we _will_ want to lose any other server-specific settings in the config file at that point, e.g. default project.

Could we conditionally merge or start with a new config from scratch?

Reply:
Ah I think I see what you mean, I will adjust this now and hopefully fix it if I am understanding correctly

Reply:
Adjusted the PR now

Reply:
Looks great! Thanks!

Reply:
Actually... my apologies... this works, but something like the following pseudocode may be more efficient and easier to read and maintain:

```
if not same server:
  cfg = new config
cfg.apiAddress = apiAddress
cfg.bearerToken = bearerToken
# etc...
```

Reply:
Agreed, that would definitely be easier to read. I've updated PR now, hopefully it makes sense - let me know if you think there's any comment(s) needed on it or if it's self-explanatory